File: apps/web/src/api/turbine.ts
import request from '@/utils/request'

export interface Turbine {
  id: number
  projectId: number
  code: string
  model: string
  lat: number
  lng: number
  altitude: number
  hubHeight: number
  status: string
  createdAt: string
  updatedAt: string
}

export interface TurbineDTO {
  projectId: number
  code: string
  model?: string
  lat: number
  lng: number
  altitude?: number
  hubHeight?: number
}

// 根据项目ID查询机位列表
export function getTurbinesByProject(projectId: number) {
  return request.get<any, Turbine[]>(`/inspection/turbines/project/${projectId}`)
}

// 获取机位详情
export function getTurbineDetail(id: number) {
  return request.get<any, Turbine>(`/inspection/turbines/${id}`)
}

// 创建机位
export function createTurbine(data: TurbineDTO) {
  return request.post<any, Turbine>('/inspection/turbines', data)
}

// 更新机位
export function updateTurbine(id: number, data: TurbineDTO) {
  return request.put<any, Turbine>(`/inspection/turbines/${id}`, data)
}

// 删除机位
export function deleteTurbine(id: number) {
  return request.delete(`/inspection/turbines/${id}`)
}

File: services/inspection-service/src/main/java/com/shanghaiwind/inspection/entity/Turbine.java
package com.shanghaiwind.inspection.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 机位实体
 */
@Data
@TableName("biz_turbine")
public class Turbine {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * 项目ID
     */
    private Long projectId;

    /**
     * 机位编号
     */
    private String code;

    /**
     * 机型
     */
    private String model;

    /**
     * 纬度
     */
    private Double lat;

    /**
     * 经度
     */
    private Double lng;

    /**
     * 海拔高度
     */
    private Double altitude;

    /**
     * 轮毂高度
     */
    private Double hubHeight;

    /**
     * 状态: PENDING, INSTALLED, COMMISSIONED
     */
    private String status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
